rotkehlchen/chain/evm/decoding/aave/v3/decoder.py

```diff
@@ -180,13 +192,39 @@ def _decode_interest(
                     event.notes = f'Receive {event.amount} {event.asset.symbol_or_name()} from migrating an AAVE v2 position to v3'  # noqa: E501
                     event.address = None  # no need to have zero address
                     migratein_event = event
+                elif swap_event is not None and swap_event.extra_data is not None:  # logic for collateral swap  # noqa: E501
+                    if (
+                        received_addr := swap_event.extra_data.get('received_addr')
+                    ) is None:
+                        log.error(
+                            f'Aave V3 swap event in {transaction} missing extradata. Skipping',
+                        )
+                        continue
+
+                    received_token = event.asset.resolve_to_evm_token()
```

Comment:
Can you maybe explain why this check is here? I don't totally get it. Discord is also fine